### PR TITLE
fix(e2e, playwright): resolved mobile test failure on action-row.spec.ts

### PR DESCRIPTION
**Pull request by rahulsuresh-git**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR is part of an effort to ensure Playwright e2e tests function correctly on mobile. (refer to #52198).

## Issue 
The playwright e2e test `action-row.spec.ts` fails on mobile browsers such as Chrome and Safari, as it tried accessing a DOM element that weren't present in the mobile view. We don't see this failure in CI since mobile tests aren't run on CI. In this test, it was observed that the 'data-playwright-test-label' is missing in the mobile view. Adding it appears to resolve the issue.

### Expected Behaviour
The test should verify the correct conditions on mobile devices and pass accordingly.

### Before fix
Screenshot 2023-11-28 at 3 39 48 PM

### After fix
Screenshot 2023-11-28 at 4 16 53 PM

### CODE REVIEW

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 1]
```diff
@@ -318,6 +318,7 @@ class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
             <div className='portal-button-wrap'>
               <button
                 className='portal-button'
+                data-playwright-test-label='preview-portal-button'
```
[the remark this answers is not in the document]
Comment (rahulsuresh-git) [reply client/src/templates/Challenges/classic/mobile-layout.tsx:321]: I see, thank you for this @huyenltnguyen. In this case, the target element is the button with the SVG icon as shown in pic. How would you suggest grabbing this using role?
image

[the remark this answers is not in the document]
Comment (rahulsuresh-git) [reply client/src/templates/Challenges/classic/mobile-layout.tsx:321]: @huyenltnguyen Do let me know if you have any thoughts around this ^ Thanks!

[the remark this answers is not in the document]
Comment (rahulsuresh-git) [reply client/src/templates/Challenges/classic/mobile-layout.tsx:321]: Oh interesting. Thank you for this! I'll get this added. @huyenltnguyen

[the remark this answers is not in the document]
Comment (rahulsuresh-git) [reply client/src/templates/Challenges/classic/mobile-layout.tsx]: @huyenltnguyen I've addressed this and verified locally. The tests are passing as expected. Please review.

**Review by huyenltnguyen — COMMENTED** [review]
Thank you for the update!

Now that we are using `getByRole()` in all tests, I hope you don't mind helping clean up the `data-playwright-test-label` attribute as well: 

https://github.com/freeCodeCamp/freeCodeCamp/blob/5679df5cdec2fc953f6df245832696eaf4578d3e/client/src/templates/Challenges/classic/action-row.tsx#L89

**Review by huyenltnguyen — APPROVED** [review]
LGTM. Thanks @rahulsuresh-git!

**Review by Ahmadkashif — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Comment by moT01** [comment]
Hey @rahulsuresh-git, I noticed you are updating your branch a lot with the "update branch" button on GitHub. You don't really need to do that unless your branch needs the updates. Each time it gets updated, the CI tests run again - which consumes some of our test runners. It's not a big deal, just letting you know that it's not usually necessary.

**Comment by moT01** [comment]
No worries. Thanks for all contributions 🎉

**Pull request merged**